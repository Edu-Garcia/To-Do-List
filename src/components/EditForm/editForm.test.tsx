import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import { EditForm } from '.'
import '@testing-library/jest-dom';

describe('Edit form Test', () => {
  it('snapshot test edit form component', () => {

    const mockTask = {
      id: '1',
      title: 'Tarefa 1',
      description: 'Descrição da tarefa 1',
      complete: true,
      user: null
    }

    const { container } = render(
      <BrowserRouter>
        <EditForm
          task={mockTask}
          setTask={jest.fn()}
          onSubmit={jest.fn()}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});