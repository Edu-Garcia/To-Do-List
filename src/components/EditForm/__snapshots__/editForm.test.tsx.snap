// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit form Test snapshot test edit form component 1`] = `
<div>
  <form>
    <div>
      <label
        for="title"
      >
        Título
      </label>
      <input
        id="title"
        name="title"
        placeholder="Adicione um título"
        required=""
        type="text"
        value="Tarefa 1"
      />
    </div>
    <div>
      <label
        for="description"
      >
        Descrição
      </label>
      <input
        id="description"
        name="description"
        placeholder="Adicione uma descrição"
        type="text"
        value="Descrição da tarefa 1"
      />
    </div>
    <div
      class="buttonsContainer"
      style="flex-direction: row;"
    >
      <button
        class="button undefined"
        type="button"
      >
        Cancelar
      </button>
      <button
        class="button edit"
        type="submit"
      >
        Alterar
      </button>
    </div>
  </form>
</div>
`;
